### PR TITLE
Don't use `--quiet` verbosity by default when running dylint

### PR DESCRIPTION
**Pull request by HCastano**
It looks like as part of #703 we added [a `--quiet` flag](https://github.com/paritytech/cargo-contract/pull/703/files#diff-46230eb6eb9a3792464a39b022f624c28ce91fed38482b30f47f483f7ef14967R344)
when running the linter. The `--output-json` flag also adds a `--quiet` flag resulting in
a failed run due to duplicate flags.

```
❮ RUSTC_BOOTSTRAP=1 cargo +stable contract build --output-json
error: The argument '--quiet' was provided more than once, but cannot be used multiple times

USAGE:
    cargo dylint [OPTIONS] [NAMES]... [-- <ARGS>...]

For more information try --help
ERROR: `"/.../bin/cargo" "dylint" "--lib=ink_linting" "--quiet" "--quiet"` failed with exit code: Some(2)
```

~~I figured that we probably don't want to run the linting step when running with
`--output-json` anyways, and by skipping that step we can avoidget this duplicate flag
conflict in the first place.~~

I've instead changed this PR to not run `dylint` with `--quiet`, and instead it'll use the verbosity
specified by the user. In your PR, @athei, you specified `--quiet` runs by default. Is there any
reason for this?

### CODE REVIEW

**Comment by athei** [comment]
Why would we not want linting when using json output? Couldn't something like ink_playground use it to show linting errors to users?

**Comment by cmichi** [comment]
>I figured that we probably don't want to run the linting step when running with `--output-json` anyways

Could you explain your thinking here? How would one enable linting for `--output-json` then?

A use-case is to have CI build a contract with `--output-json` with the intention of extracting the sizes and artifact paths. In that case you would require a way to run the linting as part of the build to ensure that the contract is fine in all ways.

**Comment by HCastano** [comment]
My thinking was along the lines of our current use case in the waterfall, which doesn't really need any linting info, but I agree we should still lint contracts

**Comment by athei** [comment]
> In your PR, @athei, you specified --quiet runs by default. Is there any reason for this?

Yes. Since we build the lint now as part of the contract build (and not when building cargo contract) we get some wonky warnings during this build. I don't really want to show those warnings. They are annoying and confusing.

Maybe you can set the default verbosity for linting to quiet?

File: src/cmd/build.rs [hunk 1]
```diff
@@ -341,7 +341,7 @@ fn exec_cargo_dylint(crate_metadata: &CrateMetadata, verbosity: Verbosity) -> Re
     };
```
[the remark this answers is not in the document]
Comment (cmichi) [reply src/cmd/build.rs:341]: I like this :+1:

**Pull request merged**